public/dmlab_so_loader.cc

```diff
@@ -42,6 +44,44 @@
 #error Must define DMLAB_SO_LOCATION dynamic library path.
 #endif
 
+#ifdef __APPLE__
+ssize_t sendfile(int out_fd, int in_fd, off_t *offset, size_t count)
```

Comment:
`sendfile` is available on BSD/MacOSX as well, e.g. https://developer.apple.com/library/archive/documentation/System/Conceptual/ManPages_iPhoneOS/man2/sendfile.2.html. It might have a slightly different signature than on Linux, but we shouldn't need to reimplement it.